R/mod_data_management.R

```diff
@@ -303,50 +316,135 @@ mod_data_management_server <- function(id, db_conn, filter_data) {
     output$download1 <- downloadHandler(
       filename = paste0("pat_data-", Sys.Date(), ".csv"),
       content = function(file) {
-        write.csv(dt_out$data, file, row.names = FALSE)
+        withProgress(message = "Downloading...", value = 0, {
+          write.csv(dt_out$data, file, row.names = FALSE)
+          incProgress(1)
+        })
       }
     )
 
-    # data module
+    # complex comments ----
+
+    output$dynamic_complex_ui <- renderUI({
+      if ((nrow(dt_out$complex_comments) > 1)) {
+        n_complex_comments <- dt_out$complex_comments |>
+          dplyr::pull(comment_txt) |>
+          length()
+
+        downloadLink(ns("complex_com"), strong(paste(n_complex_comments, "complex comments identified. click here to download them")))
+
+        # downloadButton(
+        #   # ns("complex_com"),
+        #   'complex_com',
+        #   'Complex comments',
+        #   icon = icon('comment')
+        # )
+      }
+    })
+
+    output$complex_com <- downloadHandler(
+      filename = paste0("complex_comments-", Sys.Date(), ".csv"),
+      content = function(file) {
+        withProgress(message = "Downloading...", value = 0, {
+          write.csv(dt_out$complex_comments, file, row.names = FALSE)
```

Comment:
Is csv the best format? Wouldn't Word be better?

Reply:
I am giving users the functionality to download a dataframe with all the rows with complex comments alongside the columns (including the predicted labels) for them to interrogate the data. 

Do you suggest we write only the comments to a .doc file?